lib/openstudio-standards/standards/necb/NECB2011/building_envelope.rb

```diff
@@ -292,6 +292,621 @@ def max_fwdr(hdd)
+  # Adds default construction sets, based on building 'category' & 'structure'
+  # selections. This is an alternative method to 'model_apply_construction'
+  # & 'apply_standard_construction_properties' methods.
+  #
+  # @author denis@rd2.ca
+  #
+  # @param model [OpenStudio::Model::Model] a model
+  # @param necb_hdd [Boolean] whether to rely on BTAP data or EPW stat file
+  #
+  # @return [Boolean] whether default constructions were successfully added.
+  def add_construction_sets(model, necb_hdd)
+    return false unless model.is_a?(OpenStudio::Model::Model)
+
+    tag  = "space_conditioning_category"
+    bldg = model.getBuilding
+    hdd  = get_necb_hdd18(model: model, necb_hdd: necb_hdd)
+
+    # Reset constructions.
+    model.getPlanarSurfaces.sort.each(&:resetConstruction)
+
+    # Fetch NECB-required U-factors.
+    oWallU  = max_u_necb("wall", "outdoors", hdd)
+    oRoofU  = max_u_necb("roofceiling", "outdoors", hdd)
+    oFloorU = max_u_necb("floor", "outdoors", hdd)
+    oFenU   = max_u_necb("window", "outdoors", hdd)
+    oSkyU   = max_u_necb("skylight", "outdoors", hdd)
+    oDoorU  = max_u_necb("door", "outdoors", hdd)
+    gWallU  = max_u_necb("wall", "ground", hdd)
+    gRoofU  = max_u_necb("roofceiling", "ground", hdd)
+    gFloorU = max_u_necb("floor", "ground", hdd)
+
+    # FYI: Excerpt of the 'SimpleGlazing' material generated using BTAP:
+    #
+    #   SimpleGlazing:U=0.220 SHGC=0.600, !- Name
+    #     2.2,                            !- U-Factor {W/m2-K}
+    #     0.6,                            !- Solar Heat Gain Coefficient
+    #     0.21;                           !- Visible Transmittance
+    #
+    # 1. Why is the NFRC-rated U-factor = 2.2 W/m2.K (as required by NECB2011),
+    #    yet its identifier suggests 'U=0.220' - currently unfeasible.
+    # 2. If the SHGC is 0.6, why/how can the VT be 0.21? 0.35 to 0.45 is better.
+    #
+    # ref: www.wbdg.org/resources/windows-and-glazing
+    # ref: eta-publications.lbl.gov/sites/default/files/femp-spec-sel-low-e.pdf
+    #
+    # ... for the moment, sticking to SHGC 60%, yet keeping default VT.
+    shgc = 0.60
+
+    # Fetch corresponding BTAP structure parameters.
+    category  = @structure.category
+    structure = @structure.structure
+    framing   = @structure.framing
+    cladding  = @structure.cladding
+    finish    = @structure.finish
+
+    # Uninsulated slab-on-grade, except if HDD > 6999 (Uo = 0.379 W/m2.K).
+    # If HDD < 7000, slab-on-grade insulation is instead only required along
+    # perimeter. Perimeter insulation requires special treatment, à la KIVA. The
+    # vast majority of models do not have full-height basements. Exceptions:
+    #   - LargeOffice
+    #   - LargeHotel
+    #   - Hospital
+    #
+    # The solution is temporarily OK, yet a more nuanced treatment is required
+    # for full-height basement cases. This would likely require a space-by-space
+    # treatment, based on Z-axis grade-vs-floor height difference - @todo.
```

Comment:
Depending on the complexity of future 3rd-party models to be processed by BTAP (e.g. multiple basement levels, half basement/slab-on-grade), more nuanced treatment may be needed.